src/azure-cli/azure/cli/command_modules/appservice/_params.py

```diff
@@ -460,6 +468,7 @@ def load_arguments(self, _):
                    help='Provide a string value of a Storage Account in the provided Resource Group. Or Resource ID of a Storage Account in a different Resource Group')
         c.argument('consumption_plan_location', options_list=['--consumption-plan-location', '-c'],
                    help="Geographic location where Function App will be hosted. Use `az functionapp list-consumption-locations` to view available locations.")
+        c.argument('version', options_list=['--version', '-v'], help='The functions runtime version.', arg_type=get_enum_type(HOST_VERSIONS_FUNCTIONAPP), configured_default='2')
         c.argument('runtime', help='The functions runtime stack.', arg_type=get_enum_type(set(LINUX_RUNTIMES).union(set(WINDOWS_RUNTIMES))))
         c.argument('runtime_version', help='The version of the functions runtime stack. '
                                            'Allowed values for each --runtime are: ' + ', '.join(functionapp_runtime_to_version_texts))
```

Comment:
Can we put all available versions here and delete --version? What's the difference between version 2 and 3 when choosing same runtime and runtime_version?

Reply:
`runtime_version` is the version of runtime language you're using (node, python, etc...) whereas `version` is the version of the functions host you want to use (You can make a Functions v2 App or v3 App). The two aren't related other than each functions host version only supports a subset of runtime versions. We recently GA'd functions host v3 ([more info here](https://azure.microsoft.com/en-us/updates/azure-functions-runtime-30-is-now-available/)) so we need to allow customers a choice for their host version.

Reply:
To chime in...it took me a minute to figure this out when I was first ramping up.  Maybe we scrub docs to ensure that we clearly describe the differences.

Reply:
I see. Thanks for the detailed explanation.